interviews/System Design/Key-Value Store/index.adoc

```diff
@@ -0,0 +1,31 @@
+= Key-Value Store
+:authors: Malcolm Matalka <malcolm@terrateam.io>
+:labels: interview, system design, key-value store
+:source-highlighter: highlight.js
+:toc:
+
+== Background
+
+Many Terrateam customers have asked for the ability to store data between `plan`
+and `apply` operations.  Common examples are storing Lambda builds during a plan
+so the same build can be applied or storing some information that they want to
+generate in the plan but present to the user in the apply output.  Customers
+would like Terrateam to provide this feature rather than involving another
+service.
```

Comment:
Many Terrateam customers want a way to persist data between `plan` and `apply`. The most common cases are storing a Lambda build generated during the plan so the same artifact is used during apply, or saving data created in the plan that needs to be shown to the user in the apply output. Customers prefer Terrateam to provide this capability directly rather than relying on an external service.